### PR TITLE
Add mqtt support to bsbport

### PR DESCRIPTION
**Pull request by AnDann**

### CODE REVIEW

**Comment by eku** [comment]
:+1:

**Comment by eku** [comment]
Hmm, now that you are adding MQTT support for various moduls I was asking myself if it would be better to have those files under protocols/mqtt/<component>. What do you think?

**Comment by AnDann** [comment]
I thought i leave it the same like the ecmd files which are all in same directory like the module

File: protocols/bsbport/bsbport_mqtt.c [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+#define BSBPORT_MQTT_RETAIN             false
+#define DATA_LENGTH                     8
+#define TOPIC_LENGTH                    (sizeof(BSBPORT_MQTT_TOPIC) + 17)
+
+static const char publish_topic_format[] PROGMEM = BSBPORT_PUBLISH_FORMAT;
+
+#ifdef DEBUG_BSBPORT_MQTT
+#define BSBDEBUG(s, ...) debug_printf("BSB " s "\n", ## __VA_ARGS__)
+#else
+#define BSBDEBUG(...) do { } while(0)
```
Comment (eku) [inline protocols/bsbport/bsbport_mqtt.c]: This kind of code sequence is used many times in Ethersex. I was wondering which old broken compiler requires it?

File: protocols/bsbport/bsbport_mqtt.c [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+    BSBDEBUG("MQTT set ARGS:%d %02x %02x %02x %02x %02x %3s %s %d %u", ret,
+             p1, p2, p3, p4, dest, type, strvalue, fp_val, raw_val);
+
+    if (ret == 6)
+    {
+      uint8_t data[3];
+      uint8_t datalen = 3;
+      data[0] = 0x01;
+      if (strcmp_P(type, PSTR("RAW")) == 0)
```
Comment (eku) [inline protocols/bsbport/bsbport_mqtt.c]: How about using the first charachter to indentify the type? If ambiguous, test the next character?

File: protocols/bsbport/bsbport_mqtt.c (hunk 1 above)
Comment (AnDann) [inline protocols/bsbport/bsbport_mqtt.c]: If you use something like this you get compiler warning without the "do {} while(0)"
```c
if(something)
  ...
else
  BSBDEBUG("failed");
```

**Pull request merged**